Sources/NIOHTTP2/HTTP2StreamMultiplexer.swift

```diff
@@ -75,17 +89,39 @@ public final class HTTP2StreamMultiplexer: ChannelInboundHandler, ChannelOutboun
                 self.didReadChannels.append(channel)
             }
         } else if case .headers = frame.payload {
-            let channel = MultiplexerAbstractChannel(
-                allocator: self.channel.allocator,
-                parent: self.channel,
-                multiplexer: self,
-                streamID: streamID,
-                targetWindowSize: Int32(self.targetWindowSize),
-                outboundBytesHighWatermark: self.streamChannelOutboundBytesHighWatermark,
-                outboundBytesLowWatermark: self.streamChannelOutboundBytesLowWatermark
-            )
-            self.streams[streamID] = channel
-            channel.configure(initializer: self.inboundStreamStateInitializer, userPromise: nil)
+            let channel: MultiplexerAbstractChannel
```

Comment:
Can we arrange for `MultiplexerAbstractChannel` to paper over this distinction? I think we can do it if we make the `enum` for `inboundStreamStateInitializer` something we pass to the `MultiplexerAbstractChannel` and have it end up choosing which base channel to construct.

Reply:
Yeah, that's sensible. Done.